### DISCUSSION TITLE
Commit templates per repository (maybe even sharable)

### DISCUSSION THREAD

**Discussion by nils-a** [gitextensions/gitextensions#10808] [Ideas] [answered]
I really like the *Commit templates* feature. However, some projects use different "formats" than others, regarding commit messages.

What I love to see is having *Commit templates* on a per-repository basis. Probably even shared.

I haven't looked into the current implementation, but something I could imagine is merging the settings of 
* global (i.e. current) *Commit templates*
* `.git/gitextensions.commmittemplates.json` (i.e. per repository, but inside the `.git` folder)
* `.gitextensions.commmittemplates.json` (i.e. in the root of the repository)

Are people missing this feature, too? (Or does this already exist, and I've only ever missed it?)

**Comment by RussKie** [comment, marked as answer]
I don't use templates and I can't remember seeing other developers using
those. I can't recall anyone asking or raising issues about the
functionality for quite some time either.... Which, I guess, makes you the
first 😀

If you feel passionate about this, feel free to tinker in this area and
propose an enhancement.

**Reply by nils-a** [reply]
Thanks.
I guess I'll see if I can find some time to tinker... 😀

**Comment by vbjay** [comment]
Look at how settings are stored. Understand the hierarchy of global, distributed in repo and so on. Use that to store your commit templates. Oh. I'd use guids for ids to prevent overlap between systems.
